### PR TITLE
PWGHF: Add possibility to run without event selection (for ALICE3 MC)

### PR DESCRIPTION
**Pull request by fgrosa**

### CODE REVIEW

**Comment by fgrosa** [comment]
@vkucera @ginnocen I restored the event selection, adding the option to turn it off for ALICE3 MC productions. Could you have a look? Thanks!

**Comment by ginnocen** [comment]
hi @vkucera, does it look good to you?

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx [hunk 1]
```diff
@@ -58,8 +61,71 @@ using MyTracks = soa::Join<aod::FullTracks, aod::HFSelTrack, aod::TracksExtended
 #define MY_DEBUG_MSG(condition, cmd)
 #endif
 
+/// Event selection
+struct HfTagSelCollisions {
+
+  Produces<aod::HFSelCollision> rowSelectedCollision;
+
+  Configurable<bool> doValPlots{"doValPlots", true, "fill histograms"};
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: This option does not make any plots; it fills histograms, so I would suggest to call it `fillHistograms`.
If you can fix it in the rest of the file, that would be great. :-)

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx [hunk 2]
```diff
@@ -58,8 +61,71 @@ using MyTracks = soa::Join<aod::FullTracks, aod::HFSelTrack, aod::TracksExtended
...
+    if (doValPlots && statusCollision == 0) {
+      registry.get<TH1>(HIST("hEvents"))->Fill(2);
+    }
+
+    // fill table row
+    rowSelectedCollision(statusCollision);
+  };
+
+  // no event selection in case of no event-selection task attached
+  void processNoEvSel(aod::Collision const& collision)
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: `collision` is not accessed inside the function, so can be omitted.
So I think `(aod::Collision const&)` should work.

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx [hunk 3]
```diff
@@ -58,8 +61,71 @@ using MyTracks = soa::Join<aod::FullTracks, aod::HFSelTrack, aod::TracksExtended
...
+    }
+
+    // fill table row
+    rowSelectedCollision(statusCollision);
+  };
+};
+
 /// Track selection
-struct SelectTracks {
+struct HfTagSelTrack {
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: `HfTagSelTracks`

**Review by vkucera — CHANGES_REQUESTED** [review]

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx (hunk 1 above)
Comment (fgrosa) [reply Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: Sure no problem! In the next PR I also fixed most of the variables `snake_case` in `camelCase` (I will open it soon)

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx (hunk 2 above)
Comment (fgrosa) [reply Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: Right!

File: Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx (hunk 3 above)
Comment (fgrosa) [reply Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: Done!

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
@jgrosseo @iarsene build/O2/fullCI failed because of RAM. Looks unrelated.

**Pull request merged**